src/main.cpp

```diff
@@ -3105,6 +3114,13 @@ bool ContextualCheckBlock(const CBlock& block, CValidationState& state, CBlockIn
         }
     }
 
+    // Check that all transactions are unexpired
+    BOOST_FOREACH(const CTransaction& tx, block.vtx) {
+        if (!tx.IsCoinBase() && IsExpiredTx(tx, nHeight)) {
+            return state.DoS(10, error("%s: contains an expired transaction", __func__), REJECT_INVALID, "bad-txns-expired");
```

Comment:
This should be `DoS(100, ...)`, because we are not defining expiry by time, so miners can be unequivocally held accountable for including an expired transaction.